core/src/main/java/org/apache/spark/api/plugin/SparkWorkerPlugin.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.api.plugin;
+
+import org.apache.spark.annotation.*;
+
+/**
+ * :: DeveloperApi ::
+ * A plugin to customize the creation of PySpark and SparkR workers.
+ * <p>
+ * Plugins have two optional components: One for Python subprocesses, and one for R subprocesses. Both components
+ * implement {@link ProcessBuilderPlugin}.
+ */
+@DeveloperApi
+@Evolving
+@Unstable
+public interface SparkWorkerPlugin {
+
+    /**
+     * Return the plugin to use for Python workers.
+     *
+     * @return The plugin to use for Python workers, or null if not needed.
+     */
+    ProcessBuilderPlugin pythonPlugin();
+
+    /**
+     * Return the plugin to use for R workers.
+     *
+     * @return The plugin to use for R workers, or null if not needed.
```

Comment:
I'm mirroring the `null` contract from `SparkPlugin`. Alternatively, we can provide a default implementation returning a no-op like `pb => {}`.